pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1134,11 +1149,31 @@ type WorkflowStep struct {
 	// OnExit is a template reference which is invoked at the end of the
 	// template, irrespective of the success, failure, or error of the
 	// primary template.
+	// DEPRECATED: Use ExitTemplate.template instead.
 	OnExit string `json:"onExit,omitempty" protobuf:"bytes,11,opt,name=onExit"`
+
+	// ExitTemplate hold the template and arguments which is invoked at the end of the
+	// workflow, irrespective of the success, failure, or error status of the primary template
+	ExitTemplate *ExitTemplate `json:"exitTemplate,omitempty" protobuf:"bytes,12,opt,name=exitTemplate"`
+}
+
+type ExitTemplate struct {
```

Comment:
Not sure about the name, what about ExitHook?

Reply:
can we change the name to just `LifecycleHook`?

Reply:
I prefer ExitHook, This is mainly for Exit life cycle.  Can We  rename the upper level Struct from `Hook` to `LifecycleHook` ?